### ISSUE TITLE
Porkbun API change

### ISSUE THREAD

**Issue by cristian-aldea** [ddclient/ddclient#751]
Hello,

I'm a Porkbun customer, and just today got an email saying that they will be changing their api endpoint. Will this affect the functionality of ddclient?

I think it will, from looking at the code: https://github.com/ddclient/ddclient/blob/master/ddclient.in#L6826

I think this might be as simple as changing the domain from porkbun.com to api.porkbun.com, but I'm not sure. I haven't looked into it but I might if I have time

Old API docs: https://porkbun.com/api/json/v3/documentation
New API docs: https://api.porkbun.com/api/json/v3/documentation

They seem the same to me, just a url change.

```
Hello there,

The reason you are receiving this email is because you have an API key associated with your account. In order to ensure that any apps or tools you may have that utilize our API, we wanted to let you know about some upcoming critical updates.

We know these kinds of changes are annoying but the time has come to separate the processing of incoming API commands from our website. In previous documentation the hostname porkbun.com was used for the API but we have recently updated it to api.porkbun.com and in the near future will start enforcing its use. Please update the hostname for our API to api.porkbun.com as soon as possible.


CRITICAL UPDATE DETAILS

Type: API Hostname Change

Old Value: porkbun.com

New Value: api.porkbun.com

Deadline: 2024-12-01 00:00:00 UTC
```

**Comment by cristian-aldea** [comment]
I went ahead and created a quick PR for this, feel free to review and let me know if I missed anything

**Comment by cristian-aldea** [comment]
For reference, they also mention this domain change in https://porkbun.com/api/json/v3/documentation

See the "API Hostname" section:

```
API Hostname
api.porkbun.com

Please note that porkbun.com currently works and has historically been the correct hostname for our API. However, we will be migrating away from that hostname and the API will no function using it at some point in the future.
```

**Comment by Oni** [comment]
So basically we just need to update [this line](https://github.com/ddclient/ddclient/blob/1c0ba9a1260111f2eb75e76718d218ae2ee0b29f/ddclient.in#L6731), [this line](https://github.com/ddclient/ddclient/blob/1c0ba9a1260111f2eb75e76718d218ae2ee0b29f/ddclient.in#L6826) and [this line](https://github.com/ddclient/ddclient/blob/1c0ba9a1260111f2eb75e76718d218ae2ee0b29f/ddclient.in#L6864), right?

**Comment by cristian-aldea** [comment]
> So basically we just need to update [this line](https://github.com/ddclient/ddclient/blob/1c0ba9a1260111f2eb75e76718d218ae2ee0b29f/ddclient.in#L6731), [this line](https://github.com/ddclient/ddclient/blob/1c0ba9a1260111f2eb75e76718d218ae2ee0b29f/ddclient.in#L6826) and [this line](https://github.com/ddclient/ddclient/blob/1c0ba9a1260111f2eb75e76718d218ae2ee0b29f/ddclient.in#L6864), right?

Yes, and https://github.com/ddclient/ddclient/pull/752 changes those lines. If it could be merged we can close this issue

**Comment by clarfonthey** [comment]
This change has been merged, but could a release be pushed as well so that the change is incorporated in the latest release?
